aya-ebpf-macros/src/lsm.rs

```diff
@@ -18,46 +19,64 @@ impl Lsm {
         let mut args = syn::parse2(attrs)?;
         let hook = pop_string_arg(&mut args, "hook");
         let sleepable = pop_bool_arg(&mut args, "sleepable");
+        let cgroup = pop_bool_arg(&mut args, "cgroup");
         err_on_unknown_args(&args)?;
+
         Ok(Self {
             item,
             hook,
+            cgroup,
             sleepable,
         })
     }
 
     pub(crate) fn expand(&self) -> TokenStream {
-        let Self {
-            item,
-            hook,
-            sleepable,
-        } = self;
-        let ItemFn {
-            attrs: _,
-            vis,
-            sig,
-            block: _,
-        } = item;
-        let section_prefix = if *sleepable { "lsm.s" } else { "lsm" };
-        let section_name: Cow<'_, _> = if let Some(hook) = hook {
-            format!("{}/{}", section_prefix, hook).into()
+        if self.cgroup{
```

Comment:
The `if self.cgroup { ... } else { ... }` branches are almost identical. I think we could avoid repetition by doing something like:

```rust
let section_prefix = if self.cgroup {
    "lsm_cgroup"
} else {
    "lsm"
};
let section_name = if let Some(name) = &self.hook {
    format!("{section_prefix}/{name}");
};

let fn_name = &self.item.sig.ident;
let item = &self.item;

quote! {
    #[no_mangle]
    #[link_section = #section_name]
    fn #fn_name(ctx: *mut ::core::ffi::c_void) -> i32 {
        return #fn_name(::aya_ebpf::programs::LsmContext::new(ctx));

        #item
    }
}
````